From: Peter Xu <peterx@redhat.com>
Subject: mm-hugetlb-introduce-hugetlb_walk-fix
Date: Wed, 30 Nov 2022 10:33:44 -0500

fix build with CONFIG_HUGETLB_PAGE=n, CONFIG_ARCH_WANT_HUGE_PMD_SHARE=y

Link: https://lkml.kernel.org/r/Y4d4qh0ACmo455Nj@x1n
Signed-off-by: Peter Xu <peterx@redhat.com>
Reported-by: Eric Biggers <ebiggers@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/hugetlb.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/include/linux/hugetlb.h~mm-hugetlb-introduce-hugetlb_walk-fix
+++ a/include/linux/hugetlb.h
@@ -1248,7 +1248,8 @@ __vma_shareable_flags_pmd(struct vm_area
 static inline pte_t *
 hugetlb_walk(struct vm_area_struct *vma, unsigned long addr, unsigned long sz)
 {
-#if defined(CONFIG_ARCH_WANT_HUGE_PMD_SHARE) && defined(CONFIG_LOCKDEP)
+#if defined(CONFIG_HUGETLB_PAGE) && \
+	defined(CONFIG_ARCH_WANT_HUGE_PMD_SHARE) && defined(CONFIG_LOCKDEP)
 	struct hugetlb_vma_lock *vma_lock = vma->vm_private_data;
 
 	/*
_
